app/src/page/firstRun/internal/wallet-page.js

```diff
@@ -37,32 +61,52 @@ class WalletPage extends React.PureComponent {
   }
 
   render() {
-    const { onPasswordChanged, onWalletViewLayout } = this.props;
+    const { onPasswordChanged, onWalletViewLayout, isRetrievingSync, hasSyncedWallet } = this.props;
 
-    const content = (
-      <View onLayout={onWalletViewLayout}>
-        <Text style={firstRunStyle.title}>Password</Text>
-        <Text style={firstRunStyle.paragraph}>Please enter a password to secure your account and wallet.</Text>
-        <TextInput style={firstRunStyle.passwordInput}
-          placeholder={this.state.placeholder}
-          underlineColorAndroid="transparent"
-          secureTextEntry={true}
-          value={this.state.password}
-          onChangeText={text => this.handleChangeText(text)}
-          onFocus={() => {
-            if (!this.state.password || this.state.password.length === 0) {
-              this.setState({ placeholder: '' });
-            }
-          }}
-          onBlur={() => {
-            if (!this.state.password || this.state.password.length === 0) {
-              this.setState({ placeholder: 'password' });
-            }
-          }}
-          />
-        <Text style={firstRunStyle.infoParagraph}>Note: for wallet security purposes, LBRY is unable to reset your password.</Text>
-      </View>
-    );
+    let content;
+    if (!this.state.walletReady || !this.state.hasCheckedSync || isRetrievingSync) {
+      content = (
+        <View style={firstRunStyle.centered}>
+          <ActivityIndicator size="large" color={Colors.White} style={firstRunStyle.waiting} />
+            <Text style={firstRunStyle.paragraph}>Retrieving your account information...</Text>
+        </View>
+      );
+    } else {
+      content = (
+        <View onLayout={onWalletViewLayout}>
+          <Text style={firstRunStyle.title}>Password</Text>
+          <Text style={firstRunStyle.paragraph}>
+            {hasSyncedWallet ? "Please enter the password you used to secure your wallet." :
+            "Please enter a password to secure your account and wallet."}
+          </Text>
+          <TextInput style={firstRunStyle.passwordInput}
+            placeholder={this.state.placeholder}
+            underlineColorAndroid="transparent"
+            secureTextEntry={true}
+            value={this.state.password}
+            onChangeText={text => this.handleChangeText(text)}
+            onFocus={() => {
+              if (!this.state.password || this.state.password.length === 0) {
+                this.setState({ placeholder: '' });
+              }
+            }}
+            onBlur={() => {
+              if (!this.state.password || this.state.password.length === 0) {
+                this.setState({ placeholder: 'password' });
+              }
+            }}
+            />
+            {(this.state.password && this.state.password.trim().length) > 0 &&
+              <View style={firstRunStyle.passwordStrength}>
+                <BarPasswordStrengthDisplay
+                  width={Dimensions.get('window').width - 80}
```

Comment:
That 80 seems potentially evil

Reply:
Should definitely be a constant. It's the left and right first run margins (32 + 32), and the control padding (8 + 8), added together.